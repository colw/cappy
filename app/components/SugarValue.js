import React from "react";

const directionGlyph = {
  NONE: "⇼",
  TripleUp: "⤊",
  DoubleUp: "⇈",
  SingleUp: "↑",
  FortyFiveUp: "↗",
  Flat: "→",
  FortyFiveDown: "↘",
  SingleDown: "↓",
  DoubleDown: "⇊",
  TripleDown: "⤋",
  "NOT COMPUTABLE": "-",
  "RATE OUT OF RANGE": "⇕"
};

const SugarValue = function({ sgv = 0, direction = "NONE" }) {
  return (
    <div className="main-sugar-display">
      <span className="sugar">{sgv}</span>
      <span className="direction">{directionGlyph[direction]}</span>
    </div>
  );
};

export default SugarValue;
