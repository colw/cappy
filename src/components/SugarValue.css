.main-sugar-display {
  font-size: 3rem;
}
