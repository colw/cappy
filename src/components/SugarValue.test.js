import regeneratorRuntime from "regenerator-runtime";
import "@testing-library/jest-dom";

import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import SugarValue from "./SugarValue";

describe("SugarValue", () => {
  test("renders with no value given", async () => {
    const { container, getByText } = render(<SugarValue />);
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText(/0/)).toBeInTheDocument();
    expect(getByText(/⇼/)).toBeInTheDocument();
  });

  test("renders glucose value", async () => {
    const { container, getByText } = render(
      <SugarValue sgv={123} direction="Flat" />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText(/123/)).toBeInTheDocument();
    expect(getByText(/→/)).toBeInTheDocument();
  });
});
