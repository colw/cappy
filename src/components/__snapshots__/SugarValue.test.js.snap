// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SugarValue renders glucose value 1`] = `
<div
  class="main-sugar-display"
>
  <span
    class="sugar"
  >
    123
  </span>
  <span
    class="direction"
  >
    →
  </span>
</div>
`;

exports[`SugarValue renders with no value given 1`] = `
<div
  class="main-sugar-display"
>
  <span
    class="sugar"
  >
    0
  </span>
  <span
    class="direction"
  >
    ⇼
  </span>
</div>
`;
